cleverhans/experimental/certification/optimization.py

```diff
@@ -282,6 +326,20 @@ def run_one_step(self, eig_init_vec_val, eig_num_iter_val, smooth_val,
                                 str(self.current_step) + '.json')
         with tf.gfile.Open(filename) as file_f:
           file_f.write(stats)
+
+    # Project onto feasible set of dual variables
+    if self.current_step % self.params['projection_steps'] == 0 and self.current_unconstrained_objective < 0:
+      nu = self.sess.run(self.dual_object.nu)
+      feed_dict = {
+          self.projected_dual_object.nu: nu,
+          self.dual_object.h_min_vec_ph: self.dual_object.h_min_vec_estimate,
+          self.projected_dual_object.m_min_vec_ph: self.projected_dual_object.m_min_vec_estimate
```

Comment:
it looks like `m_min_vec_estimate` is never updated and always zero in `self.projected_dual_object`

I looked at the code where `m_min_vec_estimate` is used (not in this PR) and it looks like it might better be part of `Optimization` class

Reply:
As we discussed offline, `m_min_vec_ph` intended to be zero in this case.
To achieve this I would suggest you to use `tf.placeholder_with_default`:

```
# in dual formulation code
class DualFormulation:

  def __init__():
    ...
    self.m_min_vec_ph = tf.placeholder_with_default(tf.zeros(...), ...)
```
Then you don't need to pass it into feed dict